### DISCUSSION TITLE
Release notes upload to Testflight silently failing

### DISCUSSION THREAD

**Discussion by markusunger** [codemagic-ci-cd/codemagic-docs#1703] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

Hello,

I'm trying to set up the automated release notes upload to both Testflight and Google Play for our app builds. They will be dynamically generated, so I'm relying on a per-run-generated `release_notes.json` file, but even with a hardcoded JSON file, nothing appears in the "What's new" section in Testflight, only in Google Play.

Here's what I have tried:
1. Create a dummy `release_notes.json` in the root/working directory of the project:
```
[
    {
        "language": "en-GB",
        "text": "This is a test"
    },
    {
        "language": "en-US",
        "text": "This is a test"
    },
    {
        "language": "de-DE",
        "text": "This is a test"
    }
]
```
2. Push and trigger a build via Codemagic (see build id)
3. Observe the following log entries in the `Publishing` step:
```
== Gathering artifacts ==
Reading ******** notes from "********_notes.json"
```

### Expected results

Especially with the message that the `release_notes.json` file has been found and read from, I would expect the release notes to be successfully processed and uploaded to Testflight, or - in the case of an error or problem - to see some kind of error message.

### Actual results

Instead, no further message regarding the release notes can be found in the build logs and no discernable error can be found. Testflight does not show any release notes for this build. However, they do appear in Google Play's Early Access platform.

### Build id (optional)

646dace1b37285aaef41689b

**Comment by icarusdust** [comment, marked as answer]
Hi @markusunger, first and foremost, please note that your team is on a paid plan which means that you have access to the chat widget in the UI for direct support by our team. Please use that widget next time in order to get notified directly and quicker.

If i may explain why it is happening, uploading release notes, submitting to beta review etc. actions happen at the Post-processing step (you can also call it Magic Actions introduced by Codemagic, read more about it [here](https://docs.codemagic.io/yaml-publishing/app-store-connect/#post-processing-of-app-store-connect-distribution-magic-actions and a lot more [here](https://medium.com/code-well/cut-ios-build-times-with-magic-actions-3f89e79d47b7) ) ) which means that post-processing is triggered only when there are release notes detected with the actionable whatsNew. As there is no “What’s new” in your release notes, the step is not triggered and ends up with no release note upload in Testflight

In order to solve it, you simply need to add the field in your release notes file:

```
[
    {
        "language": "en-US",
        "text": "Added new menus",
        "whatsNew": "Added new menus",
    }
]
```
